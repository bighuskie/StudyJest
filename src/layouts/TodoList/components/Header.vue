<template>
  <div class="header-wrapper">
    <div class="header-content">
      TodoList
      <input
        class="header-input"
        placeholder="TodoItem"
        data-test="input"
        v-model="inputValue"
        @keyup.enter="addItem"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="scss">
.header-wrapper {
  line-height: 60px;
  background-color: #333;
  .header-content {
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
    .header-input {
      float: right;
      width: 360px;
      margin-top: 14px;
      line-height: 24px;
      outline: none;
      color: #333;
      text-indent: 12px;
    }
  }
}
</style>
