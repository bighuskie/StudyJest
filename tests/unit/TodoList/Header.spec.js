import { shallowMount } from '@vue/test-utils'
import Header from '@/layouts/TodoList/components/Header.vue'
import { findTestWrapper } from '@/utils/testUtils.js'

describe('测试Header.vue', () => {
  it('组件样式变化做提示', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })

  it('组件中 Input 框初始值为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })

  it('组件中 Input 框输入值变化时，data数据应该同步更新', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('test')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('test')
  })

  it('组件中 Input 框输入回车，内容为空时，不外发事件', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('组件中 Input 框输入回车，内容不为空时，外发事件', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('test')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
  })

  it('组件中 Input 框输入回车，内容不为空时，外发事件，同时清空输入框', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('test')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })
})
