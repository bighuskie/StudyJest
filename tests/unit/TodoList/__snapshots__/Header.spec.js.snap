// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试Header.vue 组件样式变化做提示 1`] = `
<div class="header-wrapper">
  <div class="header-content">
    TodoList
    <input placeholder="TodoItem" data-test="input" class="header-input"></div>
</div>
`;
